bin/civi-download-tools

```diff
@@ -747,8 +755,8 @@ check_php_ext session recommended
 check_php_ext soap recommended
 check_php_ext zip recommended
 
-check_command node recommended "WARNING: Failed to locate command 'node'. NodeJS (http://nodejs.org/) is required for development of CiviCRM v4.6+."
-check_command npm recommended "WARNING: Failed to locate command 'npm'. NodeJS (http://nodejs.org/) is required for development of CiviCRM v4.6+."
+check_command node recommended "WARNING:\ Failed\ to\ locate\ command\ \"node\".\ NodeJS\ \(http://nodejs.org/\)\ is\ required\ for\ development\ of\ CiviCRM\ v4.6+."
+check_command npm recommended "WARNING:\ Failed\ to\ locate\ command\ \"npm\".\ NodeJS\ \(http://nodejs.org/\)\ is\ required\ for\ development\ of\ CiviCRM\ v4.6+."
```

Comment:
Do you have any sense for why this encoding changed? e.g. Bionic has a newer version of `bash` which makes a subtle change in syntax?

The question is partially idle/self-education, but it's also trying to double-check that we're not getting in boxed-in -- e.g. if there's a compatibility-break in the `bash` syntax, then fixing for a new interpreter could break an old interpreter.

Reply:
Bionic does run a [newer version of bash](https://packages.ubuntu.com/search?suite=all&searchon=names&keywords=bash). Unfortunately I'm not sure of the root cause here but I can confirm that the new encoding/escaping works on Debian Jessie and Stretch, and Ubuntu 14.04+

Reply:
OK, nice testing then. Debian Jessie seems like a fair benchmark. Double-check, checked. ✅

Reply:
I didn't test with Precise Pangolin (I figure it's too old and it won't install BK anyway due to a nodejs issue)

Reply:
Hrm, well, I just checked the `bash` version that's bundled in OSX 10.12 (macOS sierra; first release Sept 2016). It comes with bash 3.2 and displays the message as:

```
WARNING:\ Failed\ to\ locate\ command\ "node".\ NodeJS\ \(http://nodejs.org/\)\ is\ required\ for\ development\ of\ CiviCRM\ v4.6+.
```

Situation TBD...

Reply:
Hmmm... Didn't think of bash on OSX.

Reply:
I had been thinking of a better way to handle that message within the function which doesn't seem to suffer from the same escaping issues.

Reply:
OK, quadruple-checked, and now I'm confused. On OSX with either bash 3.2 (via apple's stock build) or bash 4.4 (via nix's build), the message displays with funny escaping.

NOTE: To test how the message is displayed, I locally hack the `civi-download-tools` so that's looking for non-existent `nodezz` instead of `node`.  I also tweak the first `#!` line to make sure it doesn't grab the wrong interpreter.

Reply:
On Ubuntu 18.04.1 without the changes - it fatals the entire install process - which confused me!

Reply:
For posterity -- we had a long debugging session after this, starting circa

https://chat.civicrm.org/civicrm/pl/jcw3ibc6s7bfxbizkyjzxonmre

TLDR: the use of `pecl install` leads to a bizarre crash if you execute within a `curl ... | bash ... --full ...` context. Confusingly, it works fine if you execute locally, and tracing this required us to follow a lot of red-herrings. The resolution is to skip that and provide a warning instead.